register the endpoint class for reflection

the artemis-jms example fails in native with:
```
2023-08-31 07:33:21,463 ERROR [io.qua.ver.cor.run.VertxCoreRecorder] (vert.x-eventloop-thread-0) Uncaught exception received by Vert.x: java.lang.RuntimeException: jakarta.resource.ResourceException: java.lang.NoSuchMethodException: org.acme.MyMessageEndpoint.onMessage(jakarta.jms.Message)
        at io.quarkiverse.ironjacamar.runtime.IronJacamarSupport.activateEndpoint(IronJacamarSupport.java:70)
        at io.quarkiverse.ironjacamar.runtime.IronJacamarRecorder.lambda$activateEndpoint$0(IronJacamarRecorder.java:83)
        at io.vertx.core.impl.future.FutureImpl$1.onSuccess(FutureImpl.java:91)
        at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
        at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
        at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:40)
        at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base@17.0.8/java.lang.Thread.run(Thread.java:833)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:807)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:210)
Caused by: jakarta.resource.ResourceException: java.lang.NoSuchMethodException: org.acme.MyMessageEndpoint.onMessage(jakarta.jms.Message)
        at org.apache.activemq.artemis.ra.inflow.ActiveMQActivation.<init>(ActiveMQActivation.java:176)
        at org.apache.activemq.artemis.ra.ActiveMQResourceAdapter.endpointActivation(ActiveMQResourceAdapter.java:176)
        at io.quarkiverse.ironjacamar.runtime.IronJacamarContainer.endpointActivation(IronJacamarContainer.java:52)
        at io.quarkiverse.ironjacamar.runtime.IronJacamarSupport.activateEndpoint(IronJacamarSupport.java:68)
        ... 16 more
Caused by: java.lang.NoSuchMethodException: org.acme.MyMessageEndpoint.onMessage(jakarta.jms.Message)
        at java.base@17.0.8/java.lang.Class.checkMethod(DynamicHub.java:1038)
        at java.base@17.0.8/java.lang.Class.getMethod(DynamicHub.java:1023)
        at io.quarkiverse.ironjacamar.runtime.endpoint.DefaultMessageEndpointFactory.isDeliveryTransacted(DefaultMessageEndpointFactory.java:46)
        at org.apache.activemq.artemis.ra.inflow.ActiveMQActivation.<init>(ActiveMQActivation.java:174)
        ... 19 more

```

--- a/deployment/src/main/java/io/quarkiverse/ironjacamar/deployment/IronJacamarProcessor.java
+++ b/deployment/src/main/java/io/quarkiverse/ironjacamar/deployment/IronJacamarProcessor.java
@@ -4,7 +4,6 @@ import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
-import java.util.Set;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
@@ -69,7 +68,9 @@ class IronJacamarProcessor {
     @BuildStep
     void additionalBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
         additionalBeans.produce(AdditionalBeanBuildItem.builder()
-                .addBeanClasses(CachedConnectionManagerProducer.class,
+                .addBeanClasses(
+                        IronJacamarSupport.class,
+                        CachedConnectionManagerProducer.class,
                         ConnectionManagerFactory.class,
                         TransactionIntegrationImpl.class)
                 .setUnremovable()
@@ -127,13 +128,14 @@ class IronJacamarProcessor {
     }
 
     @BuildStep
-    void registerApplicationScopedBeans(CombinedIndexBuildItem combinedIndexBuildItem,
-            BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+    void registerResourceEndpoints(CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
         IndexView index = combinedIndexBuildItem.getIndex();
-        Set<String> endpoints = index.getAnnotations(ResourceEndpoint.class)
+        String[] endpoints = index.getAnnotations(ResourceEndpoint.class)
                 .stream()
                 .map(annotationInstance -> annotationInstance.target().asClass().name().toString())
-                .collect(Collectors.toSet());
+                .toArray(String[]::new);
 
         // Register message endpoints as @ApplicationScoped beans
         additionalBeans.produce(AdditionalBeanBuildItem.builder()
@@ -142,12 +144,8 @@ class IronJacamarProcessor {
                 .setUnremovable()
                 .build());
 
-        // Add IronJacamarSupport bean
-        additionalBeans.produce(AdditionalBeanBuildItem.builder()
-                .addBeanClasses(IronJacamarSupport.class)
-                .setDefaultScope(DotNames.SINGLETON)
-                .setUnremovable()
-                .build());
+        // Add reflective access to message endpoints (needed for io.quarkiverse.ironjacamar.runtime.endpoint.DefaultMessageEndpointFactory#isDeliveryTransacted)
+        reflectiveClasses.produce(ReflectiveClassBuildItem.builder(endpoints).methods().build());
     }
 
     @BuildStep

